docs/installation/production/docker/primer.rst

```diff
@@ -70,8 +70,8 @@ Docker Hub
 
 .. [#f1] See `Docker and Shipping Containers: An Imperfect Analogy <https://www.ctl.io/developers/blog/post/docker-and-shipping-containers-a-useful-but-imperfect-analogy>`_
 .. [#f2] See `What is a container? | Docker <https://www.docker.com/resources/what-container>`_
-.. [#f3] See `Docker overview | Docker Documentation <https://docs.docker.com/get-started/overview/>`_
-.. [#f4] See `Dockerfile reference | Docker Documentation <https://docs.docker.com/engine/reference/builder/>`_
-.. [#f5] See `Use the Docker command line | Docker Documentation <https://docs.docker.com/engine/reference/commandline/cli/>`_
-.. [#f6] See `Docker Registry | Docker Documentation <https://docs.docker.com/registry/>`_
+.. [#f3] See `Docker overview | Docker Documentation <https://docs.docker.com/get-started/docker-overview/>`_
+.. [#f4] See `Dockerfile reference | Docker Documentation <https://docs.docker.com/reference/dockerfile/>`_
+.. [#f5] See `Use the Docker command line | Docker Documentation <https://docs.docker.com/reference/cli/docker/>`_
+.. [#f6] See `Docker Registry | Docker Documentation <https://docs.docker.com/retired/>`_
```

Comment:
The Docker Registry has been retired in favor of CNCF Distribution (hence "retired" in the link).

```suggestion
.. [#f6] See `What is a registry | Docker Documentation <https://docs.docker.com/get-started/docker-concepts/the-basics/what-is-a-registry/>`_
```